File: DSA/to_revise/75 - sortColors.py
# https://leetcode.com/problems/sort-colors/

from typing import List

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        zero = i = 0
        two = len(nums) - 1

        while i <= two:
            if nums[i] == 0:
                nums[i], nums[zero] = nums[zero], nums[i]
                zero += 1
                i += 1
            elif nums[i] == 2:
                nums[i], nums[two] = nums[two], nums[i]
                two -= 1
            else:
                i += 1

File: DSA/to_revise/75 - sortColors.cpp
// # https://leetcode.com/problems/sort-colors/

#include <vector>

class Solution {
public:
    void sortColors(std::vector<int>& nums) {
        int zero { 0 }, i { 0 };
        int two = nums.size() - 1;

        while (i <= two) {
            if (nums[i] == 0) {
                int temp = nums[i];
                nums[i] = nums[zero];
                nums[zero] = temp;
                zero++;
                i++;
            } else if (nums[i] == 2) {
                int temp = nums[i];
                nums[i] = nums[two];
                nums[two] = temp;
                two--;
            } else {
                i++;
            }
        }
    }
};